TelepresenceBot/mobile/src/main/java/com/novoda/tpbot/human/HumanSocketIOTpService.java

```diff
@@ -0,0 +1,117 @@
+package com.novoda.tpbot.human;
+
+import android.os.Handler;
+import android.os.Looper;
+
+import com.novoda.tpbot.Direction;
+import com.novoda.tpbot.Result;
+import com.novoda.tpbot.support.Observable;
+
+import java.net.MalformedURLException;
+import java.net.URISyntaxException;
+import java.net.URL;
+
+import io.socket.client.Ack;
+import io.socket.client.IO;
+import io.socket.client.Socket;
+import io.socket.client.SocketIOException;
+import io.socket.emitter.Emitter;
+
+import static io.socket.client.Socket.EVENT_CONNECT_ERROR;
+import static io.socket.client.Socket.Listener;
+
+class HumanSocketIOTpService implements HumanTpService {
+
+    private Socket socket;
+    private final Handler handler;
+
+    static HumanSocketIOTpService getInstance() {
+        return LazySingleton.INSTANCE;
+    }
+
+    private HumanSocketIOTpService() {
+        this.handler = new Handler(Looper.getMainLooper());
+    }
+
+    @Override
+    public Observable<Result> connectTo(String serverAddress) {
+        try {
+            URL url = new URL(serverAddress);
+            socket = IO.socket(url.toExternalForm());
+        } catch (MalformedURLException | URISyntaxException exception) {
+            String message = "Address should be in the format `http://[ip_address]:[port_number]`";
+            HumanSocketIOConnectionException exceptionWithUserFacingMessage = new HumanSocketIOConnectionException(message, exception);
+            return Observable.just(Result.from(exceptionWithUserFacingMessage));
+        }
+        return new SocketConnectionObservable();
+    }
+
+    private class SocketConnectionObservable extends Observable<Result> {
+
+        @Override
+        public Observable<Result> start() {
+            if (socket.connected()) {
+                notifyOf(Result.from("Already connected"));
+                return this;
+            }
+
+            socket.emit("join_as_human", "", joinSuccessAcknowledgementListener);
```

Comment:
What is `join_as_human`? A tag?
Please extract it with a meaningful constant name